helper/makeShell.nix

```diff
@@ -0,0 +1,109 @@
+{ configDir, extraShellInitCmds ? (pkgs: "") }:
+let
+  nixpkgs = (import ../pkgs/nixpkgs-pinned.nix).nixpkgs;
+  pkgs = import nixpkgs {};
+  nbPkgs = import ../pkgs { inherit pkgs; };
+  cfgDir = toString configDir;
+in
+with pkgs;
+stdenv.mkDerivation rec {
+  name = "nix-bitcoin-environment";
+
+  path = lib.makeBinPath [ nbPkgs.extra-container ];
+
+  shellHook = ''
+    export NIX_PATH="nixpkgs=${nixpkgs}:nix-bitcoin=${toString ../.}:."
+    export PATH="${path}''${PATH:+:}$PATH"
+
+    export NIX_BITCOIN_EXAMPLES_DIR="${cfgDir}"
+
+    help() {
+        echo "nix-bitcoin path: ${toString ../.}"
+        echo
+        echo "Available commands"
+        echo "=================="
+        echo "deploy"
+        echo "  Run krops-deploy and eval-config in parallel."
+        echo "  This ensures that eval failures appear quickly when deploying."
+        echo "  In this case, deployment is stopped."
+        echo
+        echo "krops-deploy"
+        echo "  Deploy your node via krops"
+        echo
+        echo "eval-config"
+        echo "  Evaluate your node system configuration"
+        echo
+        echo "generate-secrets"
+        echo "  Create secrets required by your node configuration."
+        echo "  Secrets are written to ./secrets/"
+        echo "  This function is automatically called by krops-deploy."
+        echo
+        echo "update-nix-bitcoin"
+        echo "  Fetch and use the latest version of nix-bitcoin"
+    }
+    h() { help; }
+
+    fetch-release() {
+      ${toString ./fetch-release}
+    }
+
+    update-nix-bitcoin() {
```

Comment:
This may be wrongly understood as actually updating nix-bitcoin while a restart of the nix-shell is still missing. I wonder if it's possible to simply restart the shell here.